### PR TITLE
DM-38150-main: Fix/update ticket list for v23

### PR DESCRIPTION
**Pull request by mwittgen**

### CODE REVIEW

**Comment by mwittgen** [comment]
Build can be found here: https://pipelines.lsst.io/v/tickets-DM-38150-main/releases/

File: releases/v23_0_0.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .. _release-v23-0-0:
 
 ###########################
-Release 23.0.0 (2021-12-21)
+Release 23.0.4 (2023-03-12)
```
Comment (jonathansick) [inline releases/v23_0_0.rst:4]: I just noticed this. Does this mean we're suppressing the original release date every time we do a new bug fix release? Normally the release log keeps a full history and I get that the Pipeline release documentation is a bit non standard for lots of reasons. I guess my question is: should we fix this issue?

Comment (mwittgen) [reply releases/v23_0_0.rst:4]: I will revert this then. Not sure this was intended or not updated in the past.

Comment (mwittgen) [reply releases/v23_0_0.rst]: Reverted

**Comment by jonathansick** [comment]
Thanks. I guess it's kind of an open question on how to treat the history of patch releases because the dates of all the releases are kind of lost when we just pick one. At least the ticket listings make it clear what changes are in each patch release. All this to say that some thinking about how the release notes are packaged could be a good idea. But this is definitely fine for now.

**Review by jonathansick — APPROVED** [review]

**Pull request merged**